tests/smoke-tests/src/test/java/org/apache/activemq/artemis/tests/smoke/replicationflow/SoakPagingTest.java

```diff
@@ -157,10 +158,14 @@ public void run() {
          }
 
          System.out.println("Awaiting producers...");
-         producersLatch.await();
+         if (!producersLatch.await(30000, TimeUnit.MILLISECONDS)) {
+            System.out.println("Awaiting producers timeout");
```

Comment:
-1000

Producers never finishing here would be an issue that should fail the test, not silently fail it.

Reply:
@clebertsuconic I see your point. I could return 0 if the Producers never finishing here. WDYT?

Reply:
System.exit(0) with a System.err.println notification would be ok.